feat: Remove webxdc info messages together with webxdc

Septias opened the pull request:
This PR adds the feature that webxdc info messages belonging to a wexbdc are deleted when the webxdc is deleted. This is implemented by checking every deleted message whether it has referencing messages with the `SystemMessage` `WebxdcInfoMsg`. These messages are then deleted locally, and a delete request is sent out to chat peers. Deletion could also be implemented on the receiving side of a webxdc delete, but with the current implementation, the code is nice and concise and in one place.

close #6999

Septias committed:
Extend delete queue with webxdc info messages

This allocates a vector of size #messagesToDelete which should be no
problem.

Septias committed:
add failing test

Hocuri commented on src/message.rs:
```diff
@@ -1763,6 +1795,7 @@ pub async fn delete_msgs_ex(
             "Cannot request deletion of unencrypted message for others"
         );
 
+        deleted_info_msgs.extend(get_webxdc_info_messages(context, &msg).await?);
```
This should only be done if `&msg` is a Webxdc message.

Hocuri commented on src/message.rs:
```diff
@@ -1783,22 +1816,30 @@ pub async fn delete_msgs_ex(
         );
         if let Some(chat_id) = modified_chat_ids.iter().next() {
-            let mut msg = Message::new_text("🚮".to_owned());
-            // We don't want to send deletion requests in chats w/o encryption:
-            // - These are usually chats with non-DC clients who won't respect deletion requests
-            //   anyway and display a weird trash bin message instead.
-            // - Deletion of world-visible unencrypted messages seems not very useful.
```
Did you mean to remove this comment because it's wrong/unnecessary, or accidentally?

Hocuri commented on src/message.rs:
```diff
@@ -1783,22 +1816,30 @@ pub async fn delete_msgs_ex(
+            delete_msg_locally(context, &msg).await?;
+        }
+
+        if let Some(chat_id) = modified_chat_ids.iter().next() {
+            send_delete_request(context, chat_id, &info_msg_rfc724).await?;
+        }
```
Shouldn't the delete request only be sent if `delete_for_all`?

Septias replied:
I think webxdc info messages should always be deleted

Hocuri replied:
But why delete them for everyone in the group, when the user clicked on "Delete for me" rather than "Delete for everyone"?

Septias committed:
fixes

Septias committed:
remove clone

Septias committed:
clippy

r10s commented:
> No, I have to test it on a real device. 

that should be done, building desktop is comparable easy - and testing multi device should be possible by ex-/importing a backup 

> But rfc724 mid should be be the same across devices, no?

no they are generated locally, at least this is what i read from the webxdc.rs code at a first glance.

but as said, testing makes sense here

Septias committed:
Delete info messages on delete requets

When receiving a delete request for webxdcs, delete the associated info
messages

Septias committed:
Delete webxdc info messages in sync
